src/utils/py/daos_api.py

```diff
@@ -614,105 +623,20 @@ class DaosObjClassOld(enum.IntEnum):
-    OC_S512        = 209
-    OC_S1K         = 210
-    OC_S2K         = 211
-    OC_S4K         = 212
-    OC_S8K         = 213
-    OC_SX          = 214
-    OC_RP_2G1      = 220
-    OC_RP_2G2      = 221
-    OC_RP_2G4      = 222
-    OC_RP_2G8      = 223
-    OC_RP_2G16     = 224
-    OC_RP_2G32     = 225
-    OC_RP_2G64     = 226
-    OC_RP_2G128    = 227
-    OC_RP_2G256    = 228
-    OC_RP_2G512    = 229
-    OC_RP_2G1K     = 230
-    OC_RP_2G2K     = 231
-    OC_RP_2G4K     = 232
-    OC_RP_2G8K     = 233
-    OC_RP_2GX      = 234
-    OC_RP_3G1      = 240
-    OC_RP_3G2      = 241
-    OC_RP_3G4      = 242
-    OC_RP_3G8      = 243
-    OC_RP_3G16     = 244
-    OC_RP_3G32     = 245
-    OC_RP_3G64     = 246
-    OC_RP_3G128    = 247
-    OC_RP_3G256    = 248
-    OC_RP_3G512    = 249
-    OC_RP_3G1K     = 250
-    OC_RP_3G2K     = 251
-    OC_RP_3G4K     = 252
-    OC_RP_3G8K     = 253
-    OC_RP_3GX      = 254
-    OC_RP_8G1      = 260
-    OC_RP_8G2      = 261
-    OC_RP_8G4      = 262
-    OC_RP_8G8      = 263
-    OC_RP_8G16     = 264
-    OC_RP_8G32     = 265
-    OC_RP_8G64     = 266
-    OC_RP_8G128    = 267
-    OC_RP_8G256    = 268
-    OC_RP_8G512    = 269
-    OC_RP_8G1K     = 270
-    OC_RP_8G2K     = 271
-    OC_RP_8G4K     = 272
-    OC_RP_8G8K     = 273
-    OC_RP_8GX      = 274
-    OC_RESERVED    = 1024
-    OC_RP_4G1      = 1025
-    OC_RP_4G2      = 1026
-    OC_RP_4G4      = 1027
-    OC_RP_4GX      = 1028
-
 ConvertObjClass = {
-    DaosObjClassOld.DAOS_OC_TINY_RW     : DaosObjClass.OC_S1,
-    DaosObjClassOld.DAOS_OC_SMALL_RW    : DaosObjClass.OC_S4,
-    DaosObjClassOld.DAOS_OC_LARGE_RW    : DaosObjClass.OC_SX,
-    DaosObjClassOld.DAOS_OC_R2S_RW      : DaosObjClass.OC_RP_2G1,
-    DaosObjClassOld.DAOS_OC_R2_RW       : DaosObjClass.OC_RP_2G2,
-    DaosObjClassOld.DAOS_OC_R2_MAX_RW   : DaosObjClass.OC_RP_2GX,
-    DaosObjClassOld.DAOS_OC_R3S_RW      : DaosObjClass.OC_RP_3G1,
-    DaosObjClassOld.DAOS_OC_R3_RW       : DaosObjClass.OC_RP_3G2,
-    DaosObjClassOld.DAOS_OC_R3_MAX_RW   : DaosObjClass.OC_RP_3GX,
-    DaosObjClassOld.DAOS_OC_R4S_RW      : DaosObjClass.OC_RP_4G1,
-    DaosObjClassOld.DAOS_OC_R4_RW       : DaosObjClass.OC_RP_4G2,
-    DaosObjClassOld.DAOS_OC_R4_MAX_RW   : DaosObjClass.OC_RP_4GX,
-    DaosObjClassOld.DAOS_OC_REPL_MAX_RW : DaosObjClass.OC_RP_XSF
+    DAOS_OC_TINY_RW     : DaosObjClass.OC_S1,
+    DAOS_OC_SMALL_RW    : DaosObjClass.OC_S4,
+    DAOS_OC_LARGE_RW    : DaosObjClass.OC_SX,
+    DAOS_OC_R2S_RW      : DaosObjClass.OC_RP_2G1,
+    DAOS_OC_R2_RW       : DaosObjClass.OC_RP_2G2,
+    DAOS_OC_R2_MAX_RW   : DaosObjClass.OC_RP_2GX,
+    DAOS_OC_R3S_RW      : DaosObjClass.OC_RP_3G1,
+    DAOS_OC_R3_RW       : DaosObjClass.OC_RP_3G2,
+    DAOS_OC_R3_MAX_RW   : DaosObjClass.OC_RP_3GX,
+    DAOS_OC_R4S_RW      : DaosObjClass.OC_RP_4G1,
+    DAOS_OC_R4_RW       : DaosObjClass.OC_RP_4G2,
+    DAOS_OC_R4_MAX_RW   : DaosObjClass.OC_RP_4GX,
+    DAOS_OC_REPL_MAX_RW : DaosObjClass.OC_RP_XSF
```

Comment:
Should there be conversions for the other DaosObjClassOld enumerations?
```
DAOS_OC_ECHO_TINY_RW  = 14
DAOS_OC_ECHO_R2S_RW   = 15
DAOS_OC_ECHO_R3S_RW   = 16
DAOS_OC_ECHO_R4S_RW   = 17
DAOS_OC_R1S_SPEC_RANK = 19
DAOS_OC_R2S_SPEC_RANK = 20
DAOS_OC_R3S_SPEC_RANK = 21
DAOS_OC_EC_K2P1_L32K  = 22
DAOS_OC_EC_K2P2_L32K  = 23
DAOS_OC_EC_K8P2_L1M   = 24
```
Currently if you call DaosObj.create() with any of these old enums you get a key error:
  File "/home/pahender/daos/src/utils/py/daos_api.py", line 745, in create
    cls = int(ConvertObjClass[DaosObjClassOld(objcls)])
KeyError: <DaosObjClassOld.DAOS_OC_R3S_SPEC_RANK: 21>

Reply:
from these enum definition in src/include/daos/object.h:
 * They should be removed after getting rid of all hard-coded
 * class IDs from python tests.

also, these DAOS_OC_ enums might also map to the existing OC_, similar to 
DaosObjClassOld.DAOS_OC_TINY_RW     : DaosObjClass.OC_S1
We will need to find out the mapping and add to the conversion table.

If not, then they may keep using their enum values defined in DaosObjClassOld, no conversion needed.